staging/src/k8s.io/client-go/tools/cache/listwatch.go

```diff
@@ -57,8 +58,14 @@ type Getter interface {
 
 // NewListWatchFromClient creates a new ListWatch from the specified client, resource, namespace and field selector.
 func NewListWatchFromClient(c Getter, resource string, namespace string, fieldSelector fields.Selector) *ListWatch {
+	return NewListWatchFromClientWithSelectors(c, resource, namespace, fieldSelector, labels.Everything())
+}
+
+// NewListWatchFromClientWithSelectors creates a new ListWatch from the specified client, resource, namespace, field selector, and label selector.
+func NewListWatchFromClientWithSelectors(c Getter, resource string, namespace string, fieldSelector fields.Selector, labelSelector labels.Selector) *ListWatch {
```

Comment:
How about a decorator like `WithSelectors(NewListWatchFromClient(...))`? All these `ListOptions` modifications are orthogonal to the actual `ListWatch` construction. `ListOptions` have even more fields. We don't want to add yet another constructor after this one. Decorators are more natural.

Reply:
Sounds much better. Let me rewrite this function. @sttts thanks